### PR TITLE
7903782: clang warnings and diagnostics below warning level should be logged as well

### PR DESCRIPTION
**Pull request by sundararajana**
Parser logs clang warnings and diagnostics below warning level as well.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Issue
 * [CODETOOLS-7903782](https://bugs.openjdk.org/browse/CODETOOLS-7903782): clang warnings and diagnostics below warning level should be logged as well (**Bug** - P4)

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**) ⚠️ Review applies to [24794e0b](https://git.openjdk.org/jextract/pull/255/files/24794e0b6962bebdd1bac15c1ab6327b367f6846)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract.git pull/255/head:pull/255` \
`$ git checkout pull/255`

Update a local copy of the PR: \
`$ git checkout pull/255` \
`$ git pull https://git.openjdk.org/jextract.git pull/255/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 255`

View PR using the GUI difftool: \
`$ git pr show -t 255`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/255.diff">https://git.openjdk.org/jextract/pull/255.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jextract/pull/255#issuecomment-2252547701)

### CODE REVIEW

File: src/main/java/org/openjdk/jextract/impl/Logger.java [hunk 1]
```diff
@@ -93,6 +87,33 @@ public void info(Position pos, String key, Object... args) {
         errWriter.println(msg);
     }
 
+    public void clangErr(Position pos, String msg) {
```
Comment (mcimadamore) [inline src/main/java/org/openjdk/jextract/impl/Logger.java:90]: Optional: maybe this is just a single method that takes a clang diagnostic, and renders it appropriately?

File: src/main/java/org/openjdk/jextract/impl/Logger.java [hunk 2]
```diff
@@ -120,6 +141,14 @@ public boolean hasWarnings() {
         return nWarnings > 0;
     }
 
+    public boolean hasClangErrors() {
+        return nClangErrors > 0;
+    }
+
+    public boolean hasClangWarnings() {
```
Comment (mcimadamore) [inline src/main/java/org/openjdk/jextract/impl/Logger.java:148]: Optional: do we need this?

**Review by mcimadamore — APPROVED** [review]

File: src/main/java/org/openjdk/jextract/impl/Logger.java (hunk 1 above)
Comment (sundararajana) [reply src/main/java/org/openjdk/jextract/impl/Logger.java:87]: No clang API binding related stuff in Logger so far. Better to leave it at that I think.

**Pull request closed without merging**